lib/nxdk/Makefile

```diff
@@ -13,3 +13,21 @@ CLEANRULES += clean-nxdk
 .PHONY: clean-nxdk
 clean-nxdk:
 	$(VE)rm -f $(NXDK_OBJS) $(NXDK_DIR)/lib/libnxdk.lib
+
+NXDK_AUTOMOUNT_D_SRCS := \
+	$(NXDK_DIR)/lib/nxdk/automount_d.c
+
+NXDK_AUTOMOUNT_D_OBJS = $(addsuffix .obj, $(basename $(NXDK_AUTOMOUNT_D_SRCS)))
+
+$(NXDK_DIR)/lib/libnxdk_automount_d.lib: $(NXDK_AUTOMOUNT_D_OBJS)
+
+ifneq ($(NXDK_DISABLE_AUTOMOUNT_D),y)
+main.exe: $(NXDK_DIR)/lib/libnxdk_automount_d.lib
+NXDK_LDFLAGS += -include:_automount_d_drive
```

Comment:
Why is this necessary?

Reply:
> Is this to prevent optimizing it away?

Yes. The linker only pulls in symbols that are referenced from libraries, ignoring the attribute, unfortunately. I've had issues with this before when I played around with LTO, and afaik our only option is to use linker options here.

I only noticed this issue here when I tested on my Xbox by launching from a dashboard, where `D:` isn't automatically mounted and the constructor wasn't run.